stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi

```diff
@@ -27,31 +33,34 @@ class FacadeDict(ImmutableContainer, dict[Any, Any]):
     def copy(self) -> None: ...  # type: ignore[override]
     def __reduce__(self): ...
 
-class Properties:
-    def __init__(self, data) -> None: ...
-    def __len__(self): ...
-    def __iter__(self): ...
-    def __dir__(self): ...
-    def __add__(self, other): ...
-    def __setitem__(self, key, obj) -> None: ...
-    def __getitem__(self, key): ...
-    def __delitem__(self, key) -> None: ...
-    def __setattr__(self, key, obj) -> None: ...
-    def __getattr__(self, key): ...
-    def __contains__(self, key): ...
-    def as_immutable(self): ...
-    def update(self, value) -> None: ...
-    def get(self, key, default: Any | None = ...): ...
-    def keys(self): ...
-    def values(self): ...
-    def items(self): ...
-    def has_key(self, key): ...
+class Properties(Generic[_T]):
+    def __init__(self, data: dict[str, _T]) -> None: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterable[_T]: ...
+    def __dir__(self) -> list[str]: ...
+    def __add__(self, other: Iterable[_S]) -> list[_S | _T]: ...
+    def __setitem__(self, key: str, obj: _T) -> None: ...
+    def __getitem__(self, key: str) -> _T: ...
+    def __delitem__(self, key: str) -> None: ...
+    def __setattr__(self, key: str, obj: _T) -> None: ...
+    def __getattr__(self, key: str) -> _T: ...
+    def __contains__(self, key: str) -> bool: ...
+    def as_immutable(self) -> ImmutableProperties[_T]: ...
+    def update(self, value: Iterable[_T]) -> None: ...
```

Comment:
This calls dict.update, so it should take an `Iterable[tuple[str, _T]]` or a `SupportsKeysAndGetItem[str, _T]`.